pulumi/integration_tests/__main__.py

```diff
@@ -0,0 +1,121 @@
+import os
+import sys
+from pathlib import Path
+
+sys.path.insert(0, "..")
+
+import pulumi_aws as aws
+import pulumi_nomad as nomad
+from infra import config
+from infra.autotag import register_auto_tags
+from infra.nomad_job import NomadJob, NomadVars
+from infra.quiet_docker_build_output import quiet_docker_output
+
+import pulumi
+
+
+def main() -> None:
+    ##### Preamble
+
+    stack_name = stackname_sans_prefix()
+
+    quiet_docker_output()
+
+    # These tags will be added to all provisioned infrastructure
+    # objects.
+    register_auto_tags({"grapl deployment": stack_name})
+
+    if not config.LOCAL_GRAPL:
+        # Set the nomad address. This can be either set as nomad:address in the config to support ssm port forwarding or
+        # taken from the nomad stack
+        nomad_config = pulumi.Config("nomad")
+        nomad_override_address = nomad_config.get("address")
+        # We prefer nomad:address to support overriding in the case of ssm port forwarding
+        nomad_server_stack = pulumi.StackReference(f"grapl/nomad/{stack_name}")
+        nomad_address = nomad_override_address or nomad_server_stack.require_output(
+            "address"
+        )
+        nomad_provider = nomad.Provider("nomad-aws", address=nomad_address)
+
+    ##### Actual Logic
+
+    grapl_stack = GraplStack(stack_name)
```

Comment:
I'm not sure I understand why you're putting this into an object. I get wanting to have a shorter alias for the StackReference.require_output(), but I'm not sure why you went and decided to do
`self.analyzer_bucket = output("analyzers-bucket")` etc.

Reply:
This is actually an idea originally floated by cm:
https://github.com/grapl-security/grapl/pull/1248#discussion_r742908772

basically, "have a well-typed object that describes what imports we can get"

Reply:
(I do agree it's a tad verbose, and perhaps make better sense in other contexts)

Reply:
Ah interesting. That makes sense.